changelog/12659.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fixed the issue of not displaying assertion failure differences when using the parameter '--import-mode=importlib' in pytest>=8.1 (#12659). The purpose of this import mode is to avoid modifications to `sys.path`, allowing test cases to have the same module names.
```

Comment:
No need to add the issue number, it gets added automatically; also I don't think we need to explain what the import mode is when describing the bug fix:

```suggestion
Fixed the issue of not displaying assertion failure differences when using the parameter ``--import-mode=importlib`` in pytest>=8.1.
```